src/test/scala/onera/pmlanalyzer/views/interference/InterferenceTestExtension.scala

```diff
@@ -0,0 +1,78 @@
+/** *****************************************************************************
+  * Copyright (c) 2021. ONERA
+  * This file is part of PML Analyzer
+  *
+  * PML Analyzer is free software ;
+  * you can redistribute it and/or modify it under the terms of the GNU Lesser General Public
+  * License as published by the Free Software Foundation ;
+  * either version 2 of  the License, or (at your option) any later version.
+  *
+  * PML  Analyzer is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY ;
+  * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  * See the GNU Lesser General Public License for more details.
+  *
+  * You should have received a copy of the GNU Lesser General Public License along with this program ;
+  * if not, write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
+  * **************************************************************************** */
+
+package onera.pmlanalyzer.views.interference
+
+import onera.pmlanalyzer.pml.exporters.FileManager
+import onera.pmlanalyzer.pml.operators.*
+import onera.pmlanalyzer.views.interference.operators.*
+import onera.pmlanalyzer.views.interference.operators.Analyse.ConfiguredPlatform
+
+import scala.concurrent.ExecutionContext.Implicits.*
+import scala.concurrent.Future
+import scala.io.Source
+import scala.language.postfixOps
+
+object InterferenceTestExtension {
+
+  extension (x:ConfiguredPlatform) {
+
+    def test(max: Int, expectedResultsDirectoryPath: String): Future[Seq[Seq[ScenarioComparison]]] = {
+      x.computeKInterference(List(max, x.initiators.size).min, ignoreExistingAnalysisFiles = true, verboseResultFile = false) map {
+        resultFiles => {
+          for {
+            i <- 2 to List(max, x.initiators.size).min
+            fileITF <- FileManager.extractResource(s"$expectedResultsDirectoryPath/${x.fullName}_itf_$i.txt")
+            fileFree <- FileManager.extractResource(s"$expectedResultsDirectoryPath/${x.fullName}_free_$i.txt")
+            rITFFile <- resultFiles.find(_.getName == s"${x.fullName}_itf_$i.txt")
+            rFreeFile <- resultFiles.find(_.getName == s"${x.fullName}_free_$i.txt")
+          } yield {
+            List(fileITF, fileFree)
```

Comment:
Is this test done if the model `x` has less initiators than the reference? If it has more?

Reply:
Here the reference and the test must have the same number of initiator.
I can consider adding an assert to send a fail if the sizes mismatch

Reply:
I have added assert ensuring that all references are available